kotlin-native/Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/StructRendering.kt

```diff
@@ -8,35 +8,45 @@ fun tryRenderStructOrUnion(def: StructDef): String? = when (def.kind) {
 }
 
 private fun tryRenderStruct(def: StructDef): String? {
-    val isPackedStruct = def.fields.any { !it.isAligned }
-
+    // The only case when offset starts from non-zero is a inner anonymous struct or union
+    val baseOffset = def.members.filterIsInstance<Field>().firstOrNull()?.offsetBytes ?: 0L
```

Comment:
So for
```c
struct S {
    struct {
        int x;
    };
    int y;
};
```
`immediateOffset` of the `AnonymousInnerRecord` will be negative? Is this intended?